examples/CMakeLists.txt

```diff
@@ -1,25 +1,35 @@
 option(GINKGO_BUILD_EXTLIB_EXAMPLE "Build the external-lib-interfacing with deal.II, you need to link the deal.II library." OFF)
-add_subdirectory(custom-logger)
-add_subdirectory(custom-matrix-format)
-add_subdirectory(custom-stopping-criterion)
+option(GINKGO_RUN_EXAMPLES "Run the examples." OFF)
```

Comment:
What is the intended effect of this flag? To me it looks like it only controls the generation of the run/validation targets, but doesn't cause them to be executed? I think we could also get away without this flag.